clang/lib/CConv/CheckedRegions.cpp

```diff
@@ -391,3 +399,20 @@ void CheckedRegionFinder::markChecked(CompoundStmt *S, int Localwild) {
 
   Map[Id] = IsChecked ? IS_CHECKED : IS_UNCHECKED;
 }
+
+void CheckedRegionFinder::emitCauseDiagnostic(PersistentSourceLoc *PSL) {
+  if (Emitted.find(PSL) == Emitted.end()) {
+    clang::DiagnosticsEngine &DE = Context->getDiagnostics();
+    unsigned ID = DE.getCustomDiagID(DiagnosticsEngine::Warning,
+                                     "Root cause of unchecked region: %0");
+    SourceManager &SM = Context->getSourceManager();
+    auto File = SM.getFileManager().getFile(PSL->getFileName());
+    SourceLocation SL = SM.translateFileLineCol(File, PSL->getLineNo(),
+                                                PSL->getColSNo());
+    if (SL.isValid()) {
+      auto DiagBuilder = DE.Report(SL, ID);
+      DiagBuilder.AddString("here");
```

Comment:
What is this "here" doing?

Reply:
Part of the warning string, made it more informative